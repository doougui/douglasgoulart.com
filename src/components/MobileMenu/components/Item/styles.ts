import { motion } from 'framer-motion';
import styled, { css } from 'styled-components';

export const Container = styled(motion.li)`
  ${({ theme }) => css`
    padding-bottom: 2rem;
    padding-top: 2rem;
    border-bottom: 1px solid ${theme.colors.textDarker};
  `}
`;

export const LinkText = styled.a`
  ${({ theme }) => css`
    color: ${theme.colors.text};
    font-size: ${theme.font.sizes.medium};
    line-height: ${theme.font.heights.medium};
    text-decoration: none;
  `}
`;
