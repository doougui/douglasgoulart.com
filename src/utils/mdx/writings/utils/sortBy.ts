import { Writing } from 'types/Writing';
import { parseToDate } from 'utils/time/parseToDate';
import { SortTypesKeys } from '../types';

function byNew(data: Writing[]) {
  return data.sort((a, b) => {
    const aAsDate = parseToDate(a.publishedAt);
    const bAsDate = parseToDate(b.publishedAt);
    return Number(bAsDate) - Number(aAsDate);
  });
}

function byName(data: Writing[]) {
  return data.sort((a, b) => (a.title > b.title ? 1 : -1));
}

export function sortBy(data: Writing[], type: SortTypesKeys): Writing[] {
  const sortTypes: { [key in SortTypesKeys]: () => Writing[] } = {
    new: () => byNew(data),
    name: () => byName(data),
  };

  return sortTypes[type]();
}
