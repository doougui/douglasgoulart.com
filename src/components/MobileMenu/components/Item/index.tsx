import Link from 'next/link';
import * as S from './styles';

type ItemProps = {
  href: string;
} & React.AnchorHTMLAttributes<HTMLAnchorElement>;

export function Item({ href, children, ...props }: ItemProps) {
  const variants = {
    open: {
      x: 0,
      opacity: 1,
      transition: {
        x: { stiffness: 1000, velocity: -100 },
      },
    },
    closed: {
      x: 200,
      opacity: 0,
      transition: {
        x: { stiffness: 1000 },
      },
    },
  };

  return (
    <S.Container variants={variants}>
      <Link href={href} passHref>
        <S.LinkText {...props}>{children}</S.LinkText>
      </Link>
    </S.Container>
  );
}
