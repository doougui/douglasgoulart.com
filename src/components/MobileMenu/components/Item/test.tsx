import { render, screen } from 'utils/tests';

import { Item } from '.';

describe('<Item />', () => {
  it('should render correctly', () => {
    const { container } = render(<Item href="/item">Item</Item>);

    expect(screen.getByRole('link', { name: 'Item' })).toHaveAttribute(
      'href',
      '/item',
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
