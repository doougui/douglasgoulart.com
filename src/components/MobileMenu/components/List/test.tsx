import { render, screen } from 'utils/tests';

import { List } from '.';
import { Item } from '../Item';

jest.mock('../Item', () => ({
  __esModule: true,
  Item: () => <div data-testid="item" />,
}));

describe('<List />', () => {
  it('should render correctly with children', () => {
    const { container } = render(
      <List>
        <Item href="/">Home</Item>
        <Item href="/about">About</Item>
      </List>,
    );

    expect(screen.getAllByTestId('item')).toHaveLength(2);

    expect(container.firstChild).toMatchSnapshot();
  });
});
