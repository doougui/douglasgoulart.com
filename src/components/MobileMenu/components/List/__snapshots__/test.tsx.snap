// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> should render correctly with children 1`] = `
.c0 {
  width: 100%;
  list-style: none;
  margin-top: 10rem;
  text-align: right;
}

<ul
  class="c0"
>
  <div
    data-testid="item"
  />
  <div
    data-testid="item"
  />
</ul>
`;
